src/v/kafka/latency_probe.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "config/configuration.h"
+#include "prometheus/prometheus_sanitize.h"
+#include "utils/hdr_hist.h"
+
+#include <seastar/core/metrics.hh>
+
+namespace kafka {
+class latency_probe {
+public:
+    void setup_metrics() {
+        namespace sm = ss::metrics;
+
+        if (config::shard_local_cfg().disable_metrics()) {
+            return;
+        }
+        std::vector<sm::label_instance> labels{
+          sm::label("latency_metric")("microseconds")};
+        _metrics.add_group(
+          prometheus_sanitize::metrics_name("fetch_and_produce_latency_us"),
+          {sm::make_histogram(
+             "fetch_latency_us",
+             sm::description("Fetch Latency"),
+             labels,
+             [this] { return _fetch_latency.seastar_histogram_logform(); }),
+           sm::make_histogram(
+             "produce_latency_us",
+             sm::description("Produce Latency"),
+             labels,
+             [this] { return _produce_latency.seastar_histogram_logform(); })});
+    }
+
+    std::unique_ptr<hdr_hist::measurement> auto_produce_measurement() {
+        return _produce_latency.auto_measure();
+    }
+    std::unique_ptr<hdr_hist::measurement> auto_fetch_measurement() {
+        return _fetch_latency.auto_measure();
+    }
+
+private:
+    static hdr_hist make_small_hdr_hist() {
```

Comment:
i think this could very well be the default to be honest. what's the memory footprint of this, can you print it to stdout or smth on the constructor you can get the size in bytes. 

we can move this function into the hdr hist no? make it public. use it as the default ctor too? 

what do you think? @dotnwat - we don't really have latencies expanding more than 1mi in practice.

Reply:
With the parameters I've selected, the memory usage is 8x less. I think it makes sense to use as the default parameters for the classes' constructor args

Reply:
so from 185KB -> 23KB (much better and still expensive, but not a blocker)  - can you double check by serializing it to a byte array 

btw, i don't think we need 1 microsecond granularity, we can probably get away with 10microsecond granularity

Reply:
Lowering the granularity to one with the other parameters I've chosen reduces the memory foot by a factor of 6.5x

Reply:
this is good stuff. I left my response in my review.